src/hotspot/share/services/memReporter.hpp

```diff
@@ -66,11 +66,22 @@ class MemReporterBase : public StackObj {
   }
 
   // Convert diff amount in bytes to current reporting scale
-  inline long diff_in_current_scale(size_t s1, size_t s2) const {
-    long amount = (long)(s1 - s2);
-    long scale = (long)_scale;
-    amount = (amount > 0) ? (amount + scale / 2) : (amount - scale / 2);
-    return amount / scale;
+  inline int64_t diff_in_current_scale(size_t s1, size_t s2) const {
```

Comment:
Why is this function inline?  It seems unlikely that it is performance critical.